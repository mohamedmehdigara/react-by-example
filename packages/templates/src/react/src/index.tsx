import React from 'react'
import ReactDOM from 'react-dom'
import example from './example'

ReactDOM.render(<>{example}</>, document.getElementById('root'))
