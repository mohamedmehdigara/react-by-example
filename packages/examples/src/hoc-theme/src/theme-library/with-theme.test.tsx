import { cleanup, render } from '@testing-library/react'
import React, { Component } from 'react'
import { DarkTheme, LightTheme } from './themes'
import { PropsWithTheme, withTheme } from './with-theme'

describe('withTheme', () => {
  let propsSpy: jest.Mock

  const Test = withTheme(
    class extends Component<PropsWithTheme> {
      static displayName = 'Test'

      componentDidMount = () => {
        propsSpy(this.props)
      }

      render = () => {
        return null
      }
    },
  )

  beforeEach(() => {
    propsSpy = jest.fn()
  })

  afterEach(() => {
    cleanup()
  })

  describe('dark', () => {
    beforeEach(() => {
      render(
        <DarkTheme>
          <Test />
        </DarkTheme>,
      )
    })

    it('should inject theme prop', () => {
      expect(propsSpy.mock.calls).toMatchSnapshot()
    })
  })

  describe('light', () => {
    beforeEach(() => {
      render(
        <LightTheme>
          <Test />
        </LightTheme>,
      )
    })

    it('should inject theme prop', () => {
      expect(propsSpy.mock.calls).toMatchSnapshot()
    })
  })
})
