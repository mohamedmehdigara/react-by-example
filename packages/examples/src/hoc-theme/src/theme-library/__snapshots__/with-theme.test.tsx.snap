// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withTheme dark should inject theme prop 1`] = `
Array [
  Array [
    Object {
      "theme": Object {
        "color": Object {
          "background": "#282a36",
          "foreground": "#f8f8f2",
        },
      },
    },
  ],
]
`;

exports[`withTheme light should inject theme prop 1`] = `
Array [
  Array [
    Object {
      "theme": Object {
        "color": Object {
          "background": "#f8f8f2",
          "foreground": "#282a36",
        },
      },
    },
  ],
]
`;
